### PR TITLE
feat: find broken links

### PR DESCRIPTION
**Pull request by hikchoi**
# feat: find broken links

This PR:
- adds a doctor command to cli and plugin
- plugin: finds all broken links and opens a webview with the list of broken links (if there are any)
- cli: finds all broken links and prints them as json

# Pull Request Checklist

You can go to [dendron pull requests](https://wiki.dendron.so/notes/adc39825-77a6-46cf-9c49-2642fcb4248e.html) to see full details for items in this checklist.

### Quality Assurance
- [x] Add a [test](https://wiki.dendron.so/notes/cb22bd36-d45a-4ffd-a31e-96c4b39cb7fb.html#writing-tests) for the new feature
- [x] Make sure all the existing [tests](https://wiki.dendron.so/notes/cb22bd36-d45a-4ffd-a31e-96c4b39cb7fb.html#running-all-tests) pass
- [x] Do a spot check by running your feature with our [test workspace](https://wiki.dendron.so/notes/cb22bd36-d45a-4ffd-a31e-96c4b39cb7fb.html#test-workspace)
- [x] If your tests involve running [[manual Testing|dendron://dendron.dendron-site/dendron.dev.qa.test#manual-testing]], make sure to update [[Test Workspace|dendron://dendron.dendron-site/dendron.dev.ref.test-workspace]] with any necessary notes/additions to perform manual test ^9jtWc6ov340p
- [ ] After you submit your pull request, check the output of our [integration test](https://github.com/dendronhq/dendron/actions) and make sure all tests pass
  - NOTE: If you running MacOS or Linux, pay special attention to the Windows output and vice versa if you are developing on Windows

#### Special Cases
- [~] If your tests changes an existing snapshot, make sure that snapshots are [updated](https://wiki.dendron.so/notes/cb22bd36-d45a-4ffd-a31e-96c4b39cb7fb.html#updating-test-snapshots)
- [~] If you are adding a new language feature (graphically visible in VS Code/preview/publishing), make sure that it is included in [test-workspace](https://wiki.dendron.so/notes/cb22bd36-d45a-4ffd-a31e-96c4b39cb7fb.html#test-workspace). We use this to manually inspect new changes and for auto regression testing 

### Docs
- [x] Make sure that the PR title follows our [commit style](https://wiki.dendron.so/notes/adc39825-77a6-46cf-9c49-2642fcb4248e.html#commit-style)
- [x] Please summarize the feature or impact in 1-2 lines in the PR description
- [x] If your change reflects documentation changes, also submit a PR to [dendron-site](https://github.com/dendronhq/dendron-site) and mention the doc PR link in your current PR

### Git
- [x] Make sure your branch names adhere to our commit style [[#^6zdCscSXs1MM]]
    - All PRs should start with `[feat|fix|enhance|]/[{description-of-pr-in-kebab-case}]`
        - `eg. feat/add-thisthing`

### Github Issue
- [x] If this pull request is addressing an existing issue, make sure to link this PR to the issue that it is resolving.
- [x] If the resolution comes with a document update, link the docs PR to the issue as well.

### CODE REVIEW

**Comment by hikchoi** [comment]
related: #1718

**Comment by hikchoi** [comment]
docs: https://github.com/dendronhq/dendron-site/pull/314

File: packages/engine-server/src/doctor/service.ts [hunk 1]
```diff
@@ -42,52 +44,63 @@ export class DoctorService {
...
+      return [];
+    }
+
+    const out = _.filter(links, (link) => {
+      if (link.type !== "wiki") {
+        return false;
+      }
+
+      const hasVaultPrefix = LinkUtils.hasVaultPrefix(link);
+      let vaultPrefix: DVault | undefined;
```
Comment (kevinslin) [inline packages/engine-server/src/doctor/service.ts:60]: can we rename this since its not a vaultPrefix but actually a vault?

**Review by kevinslin — CHANGES_REQUESTED** [review]

File: packages/engine-server/src/doctor/service.ts [hunk 2]
```diff
@@ -42,52 +44,63 @@ export class DoctorService {
     this.L = createLogger("DoctorService");
   }
 
-  getWildLinkDestinations(notes: NoteProps[], engine: DEngineClient) {
+  findWildLinks(note: NoteProps, notes: NoteProps[], engine: DEngineClient) {
```
[the remark this answers is not in the document]
Comment (hikchoi) [reply packages/engine-server/src/doctor/service.ts]: removed all mentions of `wild` links and switched to `broken` links.

File: packages/engine-server/src/doctor/service.ts (hunk 1 above)
Comment (hikchoi) [reply packages/engine-server/src/doctor/service.ts]: good catch. fixed.

**Review by kevinslin — APPROVED** [review]

**Pull request merged**